src/blas/highlevel.jl

```diff
@@ -1,59 +1,55 @@
 rocblas_size(t::Char, M::ROCVecOrMat) = (size(M, t=='N' ? 1 : 2), size(M, t=='N' ? 2 : 1))
 
-const ROCBLASArray{T<:ROCBLASFloat} = ROCArray{T}
-
 ###########
 #
 # BLAS 1
 #
 ###########
 
-LinearAlgebra.rmul!(x::ROCArray{<:ROCBLASFloat}, k::Number) =
+LinearAlgebra.rmul!(x::ROCArray{T}, k::Number) where T<:ROCBLASFloat =
   scal!(length(x), convert(eltype(x), k), x, 1)
 
-LinearAlgebra.rmul!(x::ROCArray{<:ROCBLASComplex}, k::Real) =
+LinearAlgebra.rmul!(x::ROCArray{T}, k::Real) where T<:ROCBLASComplex =
   scal!(length(x), convert(real(eltype(x)), k), x, 1)
 
-LinearAlgebra.rmul!(x::ROCArray{<:ROCBLASComplex}, k::Number) =
-  scal!(length(x), convert(eltype(x), k), x, 1)
```

Comment:
Because ROCBLASFloat = Union{Float32, Float64, ComplexF32, ComplexF64}` now and we already have
```julia
LinearAlgebra.rmul!(x::ROCArray{<:ROCBLASFloat}, k::Number) =
  scal!(length(x), convert(eltype(x), k), x, 1)
```